### DISCUSSION TITLE
How to unlock your bootloader

### DISCUSSION THREAD

**Discussion by salvogiangri** [salvogiangri/UN1CA#129] [Troubleshooting]
> [!CAUTION]
> Starting from One UI 8, Samsung has permanently removed the ability to unlock the bootloader.
> * If your device has officially received the update, **bootloader unlocking is no longer possible**.
> * Once you update, this change is **irreversible**: **downgrading to earlier One UI versions won't be possible anymore**.
> * **DO NOT UPDATE** to One UI 8 if you plan to unlock the bootloader.
> * **Avoid purchasing devices launched with One UI 8 or above**, as their bootloaders cannot be unlocked at all.
>
> This limitation only applies to devices that received One UI 8 officially from Samsung.

<p align="center">
  <img loading="lazy" src="https://github.com/user-attachments/assets/372ab2c7-2611-4b5a-b89c-2181a5182a85" width="30%"/>
  <img loading="lazy" src="https://github.com/user-attachments/assets/ec660551-4bff-4948-8cfe-d0633ec48e4a" width="30%"/>
</p>

Unlocking the bootloader is a fundamental step when installing a custom ROM, a custom recovery, or obtaining root access. An unlocked bootloader allows flashing and booting unofficial images on your device.

> [!WARNING]
> Unlocking the bootloader will **completely wipe all user data**. Make sure to **back up your data** before proceeding.

To check whether the bootloader is already unlocked, boot your device into **Download Mode** using your device-specific key combination and inspect the **OEM Lock** status.

The possible OEM Lock values are:
- **ON (L)**: fully locked.
- **ON (U)**: bootloader locked, OEM unlocking enabled.
- **OFF (U)**: fully unlocked.

If **no OEM Lock value** is displayed in Download Mode, your device is most likely **not unlockable due to regional or market restrictions** (commonly USA/Canada models) or due to lack of such feature (see above).

KnoxGuard should also be checked to ensure that your device has not been remotely locked by Samsung, a carrier, or an insurance provider.

Possible KnoxGuard values are the following:

- `Active`, `Locked`: your device has been remotely locked.
- `Prenormal`: your device is temporarily locked.
- `Checking`, `Completed`, `Broken`: your device is unlocked.

If KnoxGuard shows `Active`, `Prenormal` or `Locked`, installing or booting unofficial images will be **impossible**, regardless of the bootloader lock state.

To get around the `Prenormal` lock, factory reset your device and make sure to go through the initial setup **while being connected to the Internet**. If done correctly, KnoxGuard should be either `Checking` or `Completed`.

## Unlocking the bootloader
1. Open **Settings** → **About phone** → **Software information**.
2. Tap **Build number** **7 times** to enable **Developer Options**.
3. Go back to **Settings**, open **Developer Options**, and enable **OEM unlocking**.
4. Power off the device.
5. Boot into **Download Mode** using your device-specific key combination (the device may need to be connected to a PC via USB).
6. When prompted, **long‑press the volume up button** to unlock the bootloader. **This step will wipe all data and automatically reboot the device.**
7. Complete the initial setup by also making sure to **connect the device to the Internet**.
8. Re-enable **Developer Options** and verify that **OEM unlocking is present and grayed out**. This confirms that the bootloader is unlocked and that KnoxGuard has not re‑locked the device.

<p align="center">
  <img loading="lazy" src="https://github.com/user-attachments/assets/3a3c7aa7-be66-47e2-af58-ccb97924c236" width="30%"/>
</p>

If the procedure was successful, the device will display a **bootloader unlock warning** at every boot. This warning **cannot be removed** (unless the stock firmware is restored and the bootloader is re‑locked), as it is a mandatory Android security feature (source: [Google](https://source.android.com/docs/security/features/verifiedboot/boot-flow#unlocked-devices)).

## Locking the bootloader
If you decide to lock the bootloader again, follow the same steps used for unlocking, but **confirm the lock instead** when prompted in Download Mode.

Before locking the bootloader, ensure that:
- The device is running the **full, untouched stock firmware**.
- No modified partitions, kernels, or custom images are installed.

Please also note that:
* Locking the bootloader will **wipe all user data again**.
* Locking the bootloader **does NOT restore the Knox Warranty Bit**.
* If you intend to stay on stock firmware with a tripped Knox Warranty Bit, a solution such as [**KnoxPatch**](https://github.com/salvogiangri/KnoxPatch) is recommended.

**Comment by ffolleher** [comment]
### Galaxy S22 SM-S901B / One UI 8 – "Device Unlock Mode" still shown but inaccessible

I have a European Samsung Galaxy S22 (SM-S901B/DS, codename r0s) running:

- Android 16
- One UI 8
- Firmware / Bootloader: S901BXXSMGZB2
- CSC / Carrier ID: EUX (Germany)
- KG State: Completed
- Knox Warranty Bit: 0

The OEM Unlock option is completely missing from Developer Options.

ADB reports:

```text
ro.boot.flash.locked=1
ro.boot.other.locked=1
ro.boot.vbmeta.device_state=locked
ro.boot.warranty_bit=0
knox.kg.state=Completed
ro.boot.kg=0x4

Also, there is no OEM-related entry returned by:

adb shell settings list global | grep -i oem

The command returns no output.

### Interesting Download Mode behavior

When entering the initial Samsung Download Mode warning screen by holding
Volume Up + Volume Down while connecting USB, the screen still contains
Chinese text referring to a "Device Unlock Mode" by holding Volume Up.

However, holding Volume Up does NOT open the bootloader unlock confirmation
screen.

Instead, pressing Volume Up immediately proceeds to the normal Odin Download
Mode. Holding the button longer makes no difference.

I tested this multiple times, including once with Samsung Maintenance Mode
enabled.

The resulting Odin screen reports:

```text
CURRENT BINARY: Samsung Official
KG STATE: Completed (00)
Secure Download: Enabled
WARRANTY VOID: 0 (0x0000)
CARRIER_ID: EUX
```

### Questions

Has anyone investigated this specifically on the SM-S901B with the newer
U22/M bootloader?

1. Is the Chinese "Device Unlock Mode" instruction simply a leftover
   UI/resource string on One UI 8?

2. Is the unlock functionality actually removed from the bootloader,
   or is access to it only being blocked when `ro.boot.other.locked=1`?

3. Is there any known reproducible method on SM-S901B / U22/M to reach
   the genuine bootloader unlock confirmation screen?

I can provide additional read-only ADB/getprop output and test
non-destructive procedures if that would help.


<img width="692" height="1536" alt="20260920_093733" src="https://github.com/user-attachments/assets/7a2adcb1-baf6-44a9-a4f6-66436ed26e83" />

**Reply by salvogiangri** [reply]
It's literally explained in the first lines of the guide:
> > Starting from One UI 8, Samsung has permanently removed the ability to unlock the bootloader.
> > * If your device has officially received the update, **bootloader unlocking is no longer possible**.
> > * Once you update, this change is **irreversible**: **downgrading to earlier One UI versions won't be possible anymore**.
> > * **DO NOT UPDATE** to One UI 8 if you plan to unlock the bootloader.
> > * **Avoid purchasing devices launched with One UI 8 or above**, as their bootloaders cannot be unlocked at all.
> >
> > This limitation only applies to devices that received One UI 8 officially from Samsung.
